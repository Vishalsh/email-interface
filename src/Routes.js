import React from 'react';
import { Route, Switch, Redirect } from 'react-router-dom';

import routes from 'constants/routes';
import Layout from 'features/layout/Layout';
import Login from 'pages/login/Login';
import Mailbox from 'pages/mailbox/Mailbox';

const Routes = () => (
  <Switch>
    <Route exact path="/" component={Login}/>
    <Route path={routes.LOGIN} component={Login}/>

    <Layout>
      <Switch>
        <Route path={routes.DASHBOARD} component={() => <h1>DASHBOARD</h1>}/>
        <Route path={routes.LAYOUTS} component={() => <h1>LAYOUTS</h1>}/>
        <Route path={routes.GRAPHS} component={() => <h1>GRAPHS</h1>}/>
        <Route path={routes.MAILBOX} component={Mailbox}/>
        <Route path={routes.METRICS} component={() => <h1>METRICS</h1>}/>
        <Route path={routes.WIDGETS} component={() => <h1>WIDGETS</h1>}/>
        <Route path={routes.FORMS} component={() => <h1>FORMS</h1>}/>
        <Route path={routes.APP_VIEWS} component={() => <h1>APP VIEWS</h1>}/>
      </Switch>
    </Layout>
  </Switch>
);


export default Routes;
