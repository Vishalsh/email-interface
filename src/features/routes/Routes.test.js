import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Routes } from './Routes';

describe('Routes', () => {
  it('should render all the routes when user is logged in', () => {
    const props = {
      isLoggedIn: true
    };
    const component = shallow(<Routes { ...props }/>);

    expect(toJson(component)).toMatchSnapshot();
  });

  it('should not render the protected routes when user is not logged in', () => {
    const props = {
      isLoggedIn: false
    };
    const component = shallow(<Routes { ...props }/>);

    expect(toJson(component)).toMatchSnapshot();
  });
});