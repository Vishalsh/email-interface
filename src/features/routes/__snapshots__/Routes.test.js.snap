// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes should not render the protected routes when user is not logged in 1`] = `
<Switch>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    path="/login"
  />
  <Redirect
    push={false}
    to="/login"
  />
</Switch>
`;

exports[`Routes should render all the routes when user is logged in 1`] = `
<Switch>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    path="/login"
  />
  <Layout>
    <Switch>
      <Route
        component={[Function]}
        path="/dashboard"
      />
      <Route
        component={[Function]}
        path="/layouts"
      />
      <Route
        component={[Function]}
        path="/graphs"
      />
      <Route
        component={[Function]}
        path="/mailbox"
      />
      <Route
        component={[Function]}
        path="/metrics"
      />
      <Route
        component={[Function]}
        path="/widgets"
      />
      <Route
        component={[Function]}
        path="/forms"
      />
      <Route
        component={[Function]}
        path="/app-views"
      />
    </Switch>
  </Layout>
</Switch>
`;
